src/inference/transformer.py

```diff
@@ -256,38 +256,12 @@ def transform_images(self, images, shape, element_type, *args):
         return transformed_blob
 
 
-class PyTorchTransformer(Transformer):
-    def __init__(self, converting):
-        self._converting = converting
-
-    def __set_norm(self, image):
-        import torchvision
-
-        if not self._converting['norm']:
-            preprocess = torchvision.transforms.Compose([
-                torchvision.transforms.ToTensor()])
-            return preprocess(image.astype(np.float32))
-
-        preprocess = torchvision.transforms.Compose([
-            torchvision.transforms.ToTensor(),
-            torchvision.transforms.Normalize(mean=self._converting['mean'],
-                                             std=self._converting['std']),
-        ])
-
-        return preprocess(image.astype(np.float32) / 255)
+class PyTorchTransformer(TensorFlowLiteTransformer):
+    pass
```

Comment:
Подскажите, пожалуйста, почему предыдущая имплементация была заменена на наследование?
Раннее мы использовали встроенный препроцессинг, который оперировал `std`, `mean` (в таком же порядке). Теперь мы используем ручной препроцессинг с `input_scales` и в обратном порядке.
Таким образом, даже описанные параметры в [таблице](https://github.com/itlab-vision/dl-benchmark/blob/master/results/validation/validation_results_pytorch.md) валидации уже некорректны.

Подскажите, пожалуйста, есть ли возможность откатить эти изменения? Если нет, то пошарьте, пожалуйста, какие теперь должны быть параметры для классификационных моделей, обученных на ImageNET. Я испробовала разные варианты, но для `GrannySmith` упорно получаю то часы, то заколку для волос. Благодарю.

Reply:
У нас есть тикет на единый препроцессинг для всех фреймворков, что должно наоборот облегчить разработку в дальнейшем
Запустила alexnet из smoke test конфига, яблочко находит
```
$ python3 /home/zmas/code/dl-benchmark/src/inference/inference_pytorch.py -mn alexnet -b 1 -ni 1 --report_pa
th /home/zmas/code/dl-benchmark/test/smoke_test/logs/pytorch_python_benchmark_alexnet_29.08.23_10:17:32.json -i /mnt/nfs/datasets/omz/ILSVRC2012_img_val/ILSVRC2012_img_val/ILSVRC2012_val_00000023.JPEG --task classification --module torchvision.models --time 1 --input_name data --input_shapes [1,3,224,224] --mean [123.675,116.28,103.53
] --input_scale [58.395,57.12,57.375] --device CPU --precision FP32
[ INFO ] Get device for inference
[ INFO ] Inference will be executed on CPU
[ INFO ] Loading model alexnet from module
[ INFO ] Loading pretrained model
/home/zmas/code/dl-benchmark/smoke_env/lib/python3.8/site-packages/torchvision/models/_utils.py:223: UserWarning: Arguments other than a weight enum or `None` for 'weights' are deprecated since 0.13 and may be removed in the future. The current behavior is equivalent to passing `weights=AlexNet_Weights.IMAGENET1K_V1`. You can also use `weights=AlexNet_Weights.DEFAULT` to get the most up-to-date weights.
  warnings.warn(msg)
[ INFO ] Inference will be executed on baseline model
[ INFO ] Shape for input layer data: [1, 3, 224, 224]
[ INFO ] Preparing input data: ['/mnt/nfs/datasets/omz/ILSVRC2012_img_val/ILSVRC2012_img_val/ILSVRC2012_val_00000023.JPEG']
[ INFO ] Starting inference (max 1 iterations or 1 sec) on CPU
[ INFO ] Computing performance metrics
[ INFO ] Write report to /home/zmas/code/dl-benchmark/test/smoke_test/logs/pytorch_python_benchmark_alexnet_29.08.23_10:17:32.json
[ INFO ] Converting output tensor to print results
[ INFO ] Inference results
[ INFO ] Top 5 results:
[ INFO ] Result for image 1
[ INFO ]        0.4499772 Granny Smith
[ INFO ]        0.0933101 dumbbell
[ INFO ]        0.0876727 ocarina, sweet potato
[ INFO ]        0.0628702 hair slide
[ INFO ]        0.0484684 bottlecap
[ INFO ] Performance results:
{
    "iterations_num": 1,
    "execution_time": 0.028,
    "first_inference_time": 0.02759,
    "latency_avg": 0.02759,
    "latency_median": 0.02759,
    "latency_std": 0.0,
    "latency_max": 0.02759,
    "latency_min": 0.02759,
    "batch_throughput": 36.246,
    "throughput": 36.246
}
```

Reply:
О, вот другие `mean` и `input_scales`, которые я еще не пробовала. Вечером потестирую. Большое спасибо! ❤️